modules/blocks/chain.js

```diff
@@ -306,10 +306,7 @@ __private.applyTransaction = function(block, transaction, sender, cb) {
  * @todo Add description for the function
  */
 Chain.prototype.applyBlock = function(block, saveBlock, cb) {
-	// Transactions to rewind in case of error.
-	var appliedTransactions = {};
-
-	var undoUnconfirmedListStep = function(tx) {
+	var undoUnconfirmedListStep = function() {
 		return new Promise((resolve, reject) => {
```

Comment:
It doesn't have to be a promise now, the form of just function with cb looks easier. Please change.

Reply:
My understanding is that we are using promises like these because we plan to move from callbacks. This way we isolate the methods which use callbacks and in future we don't need to refactor alot.